src/index.ts

```diff
@@ -373,15 +361,30 @@ async function appBoot() {
 		const QUIT_APP_BUTTON_INDEX = 0;
 		const CANCEL_BUTTON_INDEX = 1;
 
-		const detailMessage = getQuitConfirmationMessage();
+		const messageInformation: Pick< MessageBoxSyncOptions, 'message' | 'detail' | 'type' > =
+			hasUploadingPushOperations()
+				? {
+						message: __( 'Sync is in progress' ),
+						detail: __(
+							"There's a sync operation in progress. Quitting the app will abort that operation. Are you sure you want to quit?"
+						),
+						type: 'warning',
+				  }
+				: {
+						message: __( 'Sync will continue' ),
+						detail: __(
+							'The sync process will continue on the server side even after you quit Studio. We will send you an email when it is complete.'
```

Comment:
I am wondering if `server side` is clear enough, I suggest for consideration
```
The sync process will continue on our servers even after you quit Studio. We will send you an email when it is complete.
```
But that is just a suggestion, feel free to keep the current one